cpp/src/gis/point_in_polygon.cu

```diff
@@ -0,0 +1,184 @@
+                {
+                    count++;
+                }
+            }
+        }
+        if ((count > 0) && (count % 2 == 0)) point_is_in_polygon[start_idx] = 1;
+        else point_is_in_polygon[start_idx] = 0;
+    }
+}
+
+struct point_in_polygon_functor {
+    template <typename col_type>
+    static constexpr bool is_supported()
+    {
+        return std::is_arithmetic<col_type>::value ;
+    }
+ 
+    template <typename col_type, typename std::enable_if< is_supported<col_type>() >::type* = nullptr>
+    void operator()(gdf_column const & d_poly_lats, gdf_column const & d_poly_lons,
+                    gdf_column const & d_point_lats, gdf_column const & d_point_lons,
+                    gdf_column const & d_point_is_in_polygon)
+    {
+
+        gdf_size_type min_grid_size = 0, block_size = 0;
+        CUDA_TRY( cudaOccupancyMaxPotentialBlockSize(&min_grid_size, &block_size, point_in_polygon_kernel<col_type>) );
+
+        point_in_polygon_kernel<col_type> <<< min_grid_size, block_size >>> (static_cast<col_type*>(d_poly_lats.data), 
+                static_cast<col_type*>(d_poly_lons.data), static_cast<col_type*>(d_point_lats.data), static_cast<col_type*>(d_point_lons.data), 
+                d_poly_lats.size, d_point_lats.size, static_cast<int8_t*>(d_point_is_in_polygon.data) );
+    }
+
+    template <typename col_type, typename std::enable_if< !is_supported<col_type>() >::type* = nullptr>
+    void operator()(gdf_column const & d_poly_lats, gdf_column const & d_poly_lons,
+                    gdf_column const & d_point_lats, gdf_column const & d_point_lons,
+                    gdf_column const & d_point_is_in_polygon)
+    {
+        CUDF_FAIL("Non-arithmetic operation is not supported");
+    }
+};
+    
+}   //  namespace
+
+namespace cudf {
+namespace gis {
+
+gdf_column point_in_polygon(gdf_column const & polygon_lats, gdf_column const & polygon_lons,
+                            gdf_column const & point_lats, gdf_column const & point_lons)
+{
+    cudaStream_t stream;
+    CUDA_TRY( cudaStreamCreate(&stream) );
+        
+    CUDF_EXPECTS(polygon_lats.data != nullptr && polygon_lons.data != nullptr, "polygon data cannot be empty");
+    CUDF_EXPECTS(point_lats.data != nullptr && point_lons.data != nullptr, "query points are empty");
+    CUDF_EXPECTS(polygon_lats.size == polygon_lons.size, "polygon size mismatch");
+    CUDF_EXPECTS(point_lats.size == point_lons.size, "query points size mismatch");
+    CUDF_EXPECTS(polygon_lats.dtype == polygon_lons.dtype, "polygon type mismatch");
+    CUDF_EXPECTS(polygon_lats.dtype == point_lats.dtype, "query point / polygon type mismatch");
+    CUDF_EXPECTS(point_lats.dtype == point_lons.dtype, "query points type mismatch");
+    CUDF_EXPECTS(polygon_lons.null_count == 0 && polygon_lats.null_count == 0, "polygon should not contain nulls");
+
+    gdf_column inside_polygon;
+    int8_t* data;
```

Comment:
This should be a `GDF_BOOL8`/`cudf::bool8` column, yes?

Reply:
Agree.

Reply:
Yes. Changed with cudf::bool8